src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -630,6 +630,26 @@ end
 @[simp] lemma is_cycle.not_of_nil {u : V} : ¬ (nil : G.walk u u).is_cycle :=
 λ h, h.ne_nil rfl
 
+/-! ### About paths -/
+
+instance [decidable_eq V] {u v : V} (p : G.walk u v) : decidable p.is_path :=
+by { rw is_path_def, apply_instance }
+
+lemma list.nodup.card_le {α : Type*} [fintype α] {l : list α} (h : l.nodup) :
+  l.length ≤ fintype.card α :=
+begin
+  classical,
+  rw ← list.to_finset_card_of_nodup h,
+  apply finset.card_le_univ,
+end
```

Comment:
Any thoughts on where this should go? It can't be in the same file as `list.to_finset_card_of_nodup` because it needs `fintype`.

Reply:
I've moved it into `data/fintype/basic` near another lemma about `list.nodup`.